x-pack/plugins/endpoint/common/types.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export class EndpointAppConstants {
+  static ENDPOINT_INDEX_NAME = 'endpoint-agent*';
+}
+
+export interface EndpointResultList {
+  // the endpoint restricted by the page size
+  endpoints: EndpointMetadata[];
+  // the total number of unique endpoints in the index
+  total: number;
+  // the page size requested
+  request_page_size: number;
+  // the index requested
+  request_page_index: number;
+}
+
+export interface EndpointMetadata {
```

Comment:
My last concern, @peluja1012 has a PR where types are being handled a bit differently: https://github.com/elastic/kibana/pull/55800/files#diff-a19f3a4b9567cb1b06760465a6a74d43R25

That PR is using `type` as opposed to `interface`.

Thoughts @paul-tavares @peluja1012 @nnamdifrankie @oatkiller

Reply:
@kevinlog I could be wrong, but I think that when you define a type directly as an object, the (ESLint) formater automatically converts it to an Interface.
The PR you reference is actually assigning a type to another Type (`Immutable`), thus why I think its ok there. (maybe?? 🤷‍♂ 😄 )

Reply:
@kevinlog I do not think we should be mixing this two. I prefer the interface and it is advised not to mix both. I think type is good for shorthand or derived type definition.

Reply:
got it.  still though, we discussed getting everything across the app into a common types file until we need to break things out.  Would it still make sense to follow the same pattern for defining an endpoint metadata response and an alert response?

Reply:
@kevinlog like Paul said, the linter will automatically convert types to interfaces when possible, so you probably don't need to think about it.

Reply:
@paul-tavares @oatkiller thanks for the knowledge and responses